### PR TITLE
Closes #5653: Expand migration success/failure codes; treat missing history/bookmarks db as success

### LINKED ISSUE #5653: Do not call into history/bookmarks migration code if `browser.db` doesn't exist
Currently we are calling these import functions as long as profile exists. We're seeing these issues in https://sentry.prod.mozaws.net/operations/fenix-fennec-nightly/issues/6963673/?query=is:unresolved - my assumption is that `browser.db` is simply missing, so let's just log this as a `success` (or `failure`?) reason without logging to Sentry.

### PR DESCRIPTION
**Pull request by grigoryk**
This PR does two things:
- adds and expands success/failure codes for migrations that didn't have full coverage
- treats `browser.db` file missing from disk as a 'success'.

Both of these changes needed changes to `metrics.yaml` that would trigger a data review, so I've combined them into a single PR to reduce churn a little bit.

---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by grigoryk** [comment]
# Request for data collection review form

**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

1) What questions will you answer with this data?

Insight into success/failure conditions surrounding various Fennec->Fenix migrations.

2) Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements? 

To support our engineering efforts around Fenne -> Fenix migration.

3) What alternative methods did you consider to answer these questions? Why were they not sufficient?

Alternative is not tracking this data. We'd like to have these insights to detect migration abnormalities.

4) Can current instrumentation answer these questions?

Not well. Some error conditions are identifiable via crash reports, but not all.

5) List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Firefox/Data_Collection) found on the Mozilla wiki.   

**Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.**

<table>
  <tr>
    <td>Measurement Description</td>
    <td>Data Collection Category</td>
    <td>Tracking Bug #</td>
  </tr>
  <tr>
    <td>history migration failure/success codes</td>
    <td>category 1 - technical data</td>
    <td>#5653</td>
  </tr>
<tr>
    <td>bookmarks migration failure/success codes</td>
    <td>category 1 - technical data</td>
    <td>#5653</td>
  </tr>
<tr>
    <td>gecko migration failure/success codes</td>
    <td>category 1 - technical data</td>
    <td>#5653</td>
  </tr>
<tr>
    <td>open tabs migration failure/success codes</td>
    <td>category 1 - technical data</td>
    <td>#5653</td>
  </tr>
<tr>
    <td>telemetry identifiers migration failure/success codes</td>
    <td>category 1 - technical data</td>
    <td>#5653</td>
  </tr>
</table>


6) How long will this data be collected?  Choose one of the following:
expires: 2020-10-01

7) What populations will you measure?
No filters - all locales, channels, etc.

8) If this data collection is default on, what is the opt-out mechanism for users?
Default Glean SDK opt-out mechanism.

9) Please provide a general description of how you will analyze this data.
By observing it in the migration dashboard (tbd).

10) Where do you intend to share the results of your analysis?
Internally in the fenix/android-components teams.

11) Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection? If so:
no.

File: components/support/migration/metrics.yaml [hunk 1]
```diff
@@ -73,6 +73,34 @@ migration.telemetry_identifiers:
...
+  success_reason:
+    type: counter
+    description: >
+      Why this migration succeeded. See codes in TelemetryHelpers.kt
+    send_in_pings:
+      - migration
+    bugs:
+      - https://github.com/mozilla-mobile/android-components/issues/5653
+    data_reviews:
+      - TODO
```
Comment (boek) [inline components/support/migration/metrics.yaml]: Link to the PR here 👍

Comment (grigoryk) [reply components/support/migration/metrics.yaml]: This needs to be a link to a data review approval comment, hence the TODO :)

**Comment by grigoryk** [comment]
bors r=pocmo,boek

**Pull request merged**